cli/internal/cmd/apply_test.go

```diff
@@ -167,5 +177,215 @@ func TestSkipPhases(t *testing.T) {
+		require.NoError(fh.Write(filepath.Join(constants.TerraformWorkingDir, "tfvars"), []byte("tf state")))
+	}
+
+	testCases := map[string]struct {
+		createConfig       func(require *require.Assertions, fh file.Handler)
+		createState        func(require *require.Assertions, fh file.Handler)
+		createMasterSecret func(require *require.Assertions, fh file.Handler)
+		createAdminConfig  func(require *require.Assertions, fh file.Handler)
+		createTfState      func(require *require.Assertions, fh file.Handler)
+		stdin              string
+		flags              applyFlags
+		wantPhases         skipPhases
+		wantErr            bool
+	}{
+		"gcp: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase: struct{}{},
+			},
+		},
+		"aws: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.AWS),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase: struct{}{},
+			},
+		},
+		"azure: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.Azure),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase: struct{}{},
+			},
+		},
+		"qemu: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.QEMU),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase:  struct{}{},
+				skipImagePhase: struct{}{}, // No image upgrades on QEMU
+			},
+		},
+		"no config file": {
+			createConfig:       func(require *require.Assertions, fh file.Handler) {},
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"no admin config file, but mastersecret file exists": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"no admin config file, no master secret": {
```

Comment:
I wonder how we can make this test easier to read / understand.

```suggestion
		"skip k8s+image when init was not yet called (no admin + mastersecret file)": {
```

Reply:
Tried adding some more context to the names of the testcases.